component/src/shielded_pool/state_key.rs

```diff
@@ -37,3 +37,15 @@ pub fn spent_nullifier_lookup(nullifier: &Nullifier) -> KeyHash {
 pub fn commission_amounts(height: u64) -> KeyHash {
     format!("staking/commission_amounts/{}", height).into()
 }
+
+pub fn quarantined_note_source(note_commitment: &note::Commitment) -> KeyHash {
+    format!("shielded_pool/quarantined_note_source/{}", note_commitment).into()
+}
```

Comment:
This should just be the same as `note_source`; otherwise, how can a client query the source of a note, without knowing whether or not it was already quarantined?

Reply:
Resolved: now there is just one note source, from which we can delete things.